MoreLinq/AtMost.cs

```diff
@@ -0,0 +1,64 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Jonas Nyrup. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Returns true when the number of elements in the given sequence is smaller than
+        /// or equal to the given integer.
+        /// This method throws an exception if the given integer is negative.
+        /// </summary>
+        /// <remarks>
+        /// The number of items streamed will be less than or equal to the given integer
```

Comment:
But the result is a `Boolean` so nothing is _streamed out_. I think you mean that it will _pull_ only on those many elements from the source (unless of course it's a materialised collection)? The wording is a little confusing & unusual.